designs/Verify-Verbosity.md

```diff
@@ -0,0 +1,246 @@
+
+The provided `--verbosity` value will include to NuGet log messages with the following levels:
+
+​             | `q[uiet]` | `m[inimal]` | `n[ormal]` | `d[etails]` | `diag[nostic]`
+------------- | --------- | ----------- | ---------- | ----------- | --------------
+`Error`       | ✔️        | ✔️         | ✔️         | ✔️         | ✔️   
+`Warning`     | ✔️        | ✔️         | ✔️         | ✔️         | ✔️   
+`Minimal`     | ❌        | ✔️         | ✔️         | ✔️         | ✔️   
+`Information` | ❌        | ❌         | ✔️         | ✔️         | ✔️   
+`Verbose`     | ❌        | ❌         | ❌         | ✔️         | ✔️   
+`Debug`       | ❌        | ❌         | ❌         | ✔️         | ✔️   
+
+Note that MSBuild itself has the following mapping for it's own "log levels"
+([source](https://docs.microsoft.com/en-us/visualstudio/msbuild/obtaining-build-logs-with-msbuild?view=vs-2019#verbosity-settings)):
+
+​                                     | `q[uiet]` | `m[inimal]` | `n[ormal]` | `d[etailed]` | `diag[nostic]`
+------------------------------------- | --------- | ----------- | ---------- | ----------- | --------------
+Errors                                | ✔️        | ✔️         | ✔️         | ✔️         | ✔️   
+Warnings                              | ✔️        | ✔️         | ✔️         | ✔️         | ✔️   
+High-importance Messages              | ❌        | ✔️         | ✔️         | ✔️         | ✔️   
+Normal-importance Messages            | ❌        | ❌         | ✔️         | ✔️         | ✔️   
+Low-importance Messages               | ❌        | ❌         | ❌         | ✔️         | ✔️   
+Additional MSBuild-engine information | ❌        | ❌         | ❌         | ❌         | ✔️   
+
+In other words, you can think of the following log concepts per row as equivalent.
+
+NuGet log level    | MSBuild verbosity switch | MSBuild message type
+------------------ | ------------------------ | --------------------
+`Error`, `Warning` | `q[uiet]`                | Errors and Warnings
+`Minimal`          | `m[inimal]`              | High-importance Messages
+`Information`      | `n[ormal]`               | Normal-importance Messages
+`Verbose`, `Debug` | `d[etailed]`             | Low-importance Messages
+​                   | `diag[nostic]`           | Additional MSBuild-engine information
+
+### Example output
+
+This is output of the invocation of `dotnet nuget verify nuget.common.5.9.0-preview.2.nupkg`
+
+#### Default verbosity | `--verbosity mimimal`
+
+```
+Verifying NuGet.Common.5.9.0-preview.2
+Signature type: Author
+Verifying the author primary signature with certificate:
+  SHA256 hash: 3F9001EA83C560D712C24CF213C3D312CB3BFF51EE89435D3430BD06B5D0EECE
+  Valid from: 2/25/2018 4:00:00 PM to 1/27/2021 4:00:00 AM
+Timestamp: 11/17/2020 7:23:10 AM
+Verifying author primary signature's timestamp with timestamping service certificate:
```

Comment:
This seems redundant imo. 

Is this a `progress` log message, or only at the end?

Either way, I don't think it's log level minimal.

Reply:
Redundant to what, the author signature? If so, they are different. Once the author signs their package (creating the author signature), a trusted timestamp server then stamps the signature with a "Yup I saw this content at timestamp `11/17/2020 7:23:10 AM`". The trusted timestamp server has its own certificate with its own hash, expiration, and validity period.

Reply:
My only comment is about this specific line `Verifying author primary signature's timestamp with timestamping service certificate:`. 

That seems like a progress message, like the rest of the output is the actual result.

Reply:
Ah gotcha. We need some way to distinguish all the certificates, so some sort of "header" is nice. I don't feel strongly about the wording though :)

Reply:
Removed `Verifying author primary signature's timestamp with timestamping service certificate: `log message from `minimal` verbosity output.